### DISCUSSION TITLE
Cleanup INavigationAware view model OnDialogClosed

### DISCUSSION THREAD

**Discussion by sylvesp** [PrismLibrary/Prism#2794] [General]
I have a dialog with a region.
When the dialog closes I need to do some cleanup in the INavigationAware view models.
Seeminly OnNavigatedFrom() doesn't fire when the dialog is closing.

How can I notify my INavigationAware view models to do their cleanup?

I can solve it with by sending them an event using the IEventAggregator but I am hoping for a simpler, more elegant solution.
Any ideas?

**Comment by brianlagunas** [comment]
If the dialog has the region defined, then just get the views within the region using the RegionManager and loop through them manually and call the interface.
